bread/layout/components/form.py

```diff
@@ -391,17 +368,14 @@ def extract_attrs(data):
         or WIDGET_MAPPING[type(field.field.widget)]
     )
     if isinstance(fieldtype, type) and issubclass(fieldtype, hg.BaseElement):
-        if issubclass(fieldtype, TextInput):
-            return fieldtype(**kwargs)
-        else:
-            ret = fieldtype(
-                fieldname=field.name,
-                widgetattributes=attrs,
-                **elementattributes,
-            )
+        ret = fieldtype(
+            fieldname=field.name,
+            widgetattributes=widgetattributes,
+            boundfield=field,
+            **elementattributes,
+        )
     else:
-        ret = fieldtype
-    ret.boundfield = field
+        raise Exception(f"Could not find a widget for field {field.name}")
```

Comment:
I am not sure raising an Exception is the best behaviour here. The point is that the framework becomes immediately unusable for any project which uses some 3rd-party widget or input field. Basically the same as PhoneNumberInput, where we had to add specialized behaviour. In general can go with two strategies here:

1. Raising an exception and and force the according developer to open an issue. We could then add the implementation for the widget somewhere in the code, would maybe need to make it a bit clearer which widgets are django-native and which belong to 3rd-party packages.
2. Having a half-dezent generalized solution which displays at least *something*. The widget might not work correctly, but at least it would show something and not break the application.

I agree with the idea that using an unsupported widget is an exception here. But for the sake of developer experience (people using this code) I tend to prefer 2.